search-service/src/main/kotlin/com/egm/stellio/search/csr/service/ContextSourceUtils.kt

```diff
@@ -28,22 +27,21 @@ object ContextSourceUtils {
     fun mergeEntities(
         localEntity: CompactedEntity?,
         remoteEntitiesWithCSR: List<CompactedEntityWithCSR>
-    ): IorNel<NGSILDWarning, CompactedEntity?> = iorNel {
-        if (localEntity == null && remoteEntitiesWithCSR.isEmpty()) return@iorNel null
+    ): IorNel<NGSILDWarning, CompactedEntity?> {
+        if (localEntity == null && remoteEntitiesWithCSR.isEmpty()) return Ior.Right(null)
 
         val mergedEntity: MutableMap<String, Any> = localEntity?.toMutableMap() ?: mutableMapOf()
 
-        remoteEntitiesWithCSR.sortedBy { (_, csr) -> csr.isAuxiliary() }
-            .forEach { (entity, csr) ->
-                mergedEntity.putAll(
-                    getMergeNewValues(mergedEntity, entity, csr).toIor().toIorNel().bind()
-                )
-            }
+        val warnings = remoteEntitiesWithCSR.sortedBy { (_, csr) -> csr.isAuxiliary() }
+            .mapNotNull { (entity, csr) ->
+                getMergeNewValues(mergedEntity, entity, csr)
+                    .onRight { mergedEntity.putAll(it) }.leftOrNull()
+            }.toNonEmptyListOrNull()
 
-        mergedEntity.toMap()
+        return if (warnings == null) Ior.Right(mergedEntity) else Ior.Both(warnings, mergedEntity)
     }
 
-    private fun getMergeNewValues(
+    fun getMergeNewValues(
```

Comment:
if it is just to be able to call it in the test, you can instead make it `internal`